src/main/java/com/verily/genomewarp/GenomeWarpSerial.java

```diff
@@ -717,31 +720,43 @@ public static void main(String[] args) {
     Fasta queryFasta = null, targetFasta = null;
 
     // Solve Issue #2 - Check if VCS is a gVCF and if so extract 1) variant-only VCF, 2) BED file
-    logger.log(Level.INFO, "Checking and processing gVCF");
-    boolean haveGvcf = false;
-    final String vcfFromGvcf = ARGS.workDir + File.separator + "from_gvcf.vcf";
-    final String bedFromGvcf = ARGS.workDir + File.separator + "from_gvcf.bed";
-    try {
-      haveGvcf = GvcfToVcfAndBed.saveVcfAndBedFromGvcf(ARGS.rawQueryVcf, vcfFromGvcf, bedFromGvcf);
-    } catch (IOException ex) {
-      fail("failed to read gVCF/write VCF or BED files: " + ex.getMessage());
-    }
+    boolean haveGvcf = ARGS.rawQueryGvcf != null;
+    final String queryVcfToProcess;
+    final String queryBedToProcess;
     if (haveGvcf) {
+      queryVcfToProcess = ARGS.workDir + File.separator + "from_gvcf.vcf";
+      queryBedToProcess = ARGS.workDir + File.separator + "from_gvcf.bed";
+      if (ARGS.rawQueryVcf != null || ARGS.rawQueryBed != null) {
```

Comment:
Nit: I think the flow is a little clearer if you pull the flag validation performed here and in lines 741-744 above all this processing rather than interleaving it with the computation. E.g.

// Query input flag validation.
if (ARGS.rawQueryGvcf != null && (ARGS.rawQueryVcf != null || ARGS.rawQueryBed != null)) {
  fail("Arguments (--raw_query_vcf, --raw_query_bed) and --raw_query_gvcf are mutually exclusive");
}
if ((ARGS.rawQueryVcf == null) != (ARGS.rawQueryBed == null)) {
  fail("Either both or neither of --raw_query_vcf and --raw_query_bed must be specified");
}
if (ARGS.rawQueryGvcf == null && ARGS.rawQueryVcf == null) {
  fail("Either (--raw_query_vcf, --raw_query_bed) or --raw_query_gvcf must be specified");
}

Then you can safely operate with the haveGvcf boolean defined as is.